File: Java/games/numberguessing/numberguessing.java
import java.util.Random;
import java.util.Scanner;

public class NumberGuessingGame {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Random random = new Random();

        // Zufallszahl zwischen 1 und 100 generieren
        int numberToGuess = random.nextInt(100) + 1;
        int userGuess = 0;
        int numberOfTries = 0;

        System.out.println("Willkommen zum Zahlenraten-Spiel!");
        System.out.println("Ich habe eine Zahl zwischen 1 und 100 gewählt. Versuche, sie zu erraten.");

        // Solange der Benutzer die Zahl nicht errät
        while (userGuess != numberToGuess) {
            System.out.print("Gib deine Schätzung ein: ");
            userGuess = scanner.nextInt();
            numberOfTries++;

            // Eingabe des Benutzers auswerten
            if (userGuess < numberToGuess) {
                System.out.println("Die gesuchte Zahl ist größer.");
            } else if (userGuess > numberToGuess) {
                System.out.println("Die gesuchte Zahl ist kleiner.");
            } else {
                System.out.println("Herzlichen Glückwunsch! Du hast die Zahl " + numberToGuess + " erraten.");
                System.out.println("Du hast " + numberOfTries + " Versuche gebraucht.");
            }
        }

        scanner.close();
    }
}

File: JavaScript/games/numberguessing/numberguessing.js
// Zufallszahl zwischen 1 und 100 generieren
let number = Math.floor(Math.random() * 100) + 1;
let guess = 0;
let attempts = 0;

alert("Willkommen zum Zahlenratespiel!");
alert("Ich habe mir eine Zahl zwischen 1 und 100 ausgedacht. Versuche sie zu erraten!");

// Spielschleife
while (guess !== number) {
    // Eingabe des Benutzers abfragen
    guess = parseInt(prompt("Gib deine Schätzung ein:"));

    // Überprüfen, ob die Eingabe eine gültige Zahl ist
    if (isNaN(guess)) {
        alert("Bitte gib eine gültige Zahl ein.");
        continue;
    }

    attempts++; // Versuche erhöhen

    // Feedback
    if (guess < number) {
        alert("Zu niedrig! Versuch es noch einmal.");
    } else if (guess > number) {
        alert("Zu hoch! Versuch es noch einmal.");
    } else {
        alert(`Herzlichen Glückwunsch! Du hast die Zahl ${number} in ${attempts} Versuchen erraten.`);
    }
}
